pkg/jmxfetch/jmxfetch.go

```diff
@@ -277,15 +260,11 @@ func (j *JMXFetch) Stop() error {
 			}
 		case <-stopChan:
 		}
-
+		return nil
 	} else {
 		// Windows
-		if err := ioutil.WriteFile(j.exitFilePath, nil, 0644); err != nil {
-			log.Warnf("Could not signal JMXFetch to exit, killing instead: %v", err)
-			return j.cmd.Process.Kill()
-		}
+		return j.cmd.Process.Kill()
```

Comment:
Calls `syscall.OpenProcess(syscall.PROCESS_TERMINATE, false, uint32(pid))` under the hood which should be fine. I believe this doesn't wait for the process to be dead so we can't be 100% sure it'll succeed. Please correct me if I'm wrong. Should we maybe follow the same pattern we use for non-windows where we wait for the process to be dead?

Reply:
The non-windows logic sends a `SIGTERM`, waits and finally sends a `SIGKILL` if the process didn't end, but it doesn't wait after the `SIGKILL`. Here we send a `SIGKILL` directly, so I don't think it's worth adding a wait: there isn't much you can do if SIGKILL doesn't kill the process anyway (ie: after the wait times out).